openmc/surface.py

```diff
@@ -1662,31 +1682,42 @@ def r2(self, r2):
         check_type('r^2 coefficient', r2, Real)
         self._coefficients['r2'] = r2
 
-    def translate(self, vector):
-        """Translate surface in given direction
-
-        Parameters
-        ----------
-        vector : iterable of float
-            Direction in which surface should be translated
-
-        Returns
-        -------
-        openmc.Cone
-            Translated surface
-
-        """
-        vx, vy, vz = vector
-        if vx == 0.0 and vy == 0.0 and vz == 0.0:
-            return self
-        else:
-            x0 = self.x0 + vx
-            y0 = self.y0 + vy
-            z0 = self.z0 + vz
-            return type(self)(x0=x0, y0=y0, z0=z0, r2=self.r2)
-
-
-class XCone(Cone):
+    @dx.setter
+    def dx(self, dx):
+        check_type('dx coefficient', dx, Real)
+        self._coefficients['dx'] = dx
+
+    @dy.setter
+    def dy(self, dy):
+        check_type('dy coefficient', dy, Real)
+        self._coefficients['dy'] = dy
+
+    @dz.setter
+    def dz(self, dz):
+        check_type('dz coefficient', dz, Real)
+        self._coefficients['dz'] = dz
+
+    def _get_base_coeffs(self):
+        x0, y0, z0, r2 = self.x0, self.y0, self.z0, self.r2
+        dx, dy, dz = self.dx, self.dy, self.dz
+        cos2 = 1 / (1 + r2)
+        c1 = (dx*dx + dy*dy + dz*dz)*cos2
+
+        a = dx*dx - c1
+        b = dy*dy - c1
+        c = dz*dz - c1
+        d = 2*dx*dy
+        e = 2*dy*dz
+        f = 2*dx*dz
+        g = -2*(dx*dx*x0 + dx*dy*y0 + dx*dz*z0 - c1)
+        h = -2*(dy*dy*y0 + dx*dy*x0 + dy*dz*z0 - c1)
+        j = -2*(dz*dz*y0 + dx*dz*x0 + dy*dz*y0 - c1)
+        k = (dx*x0 + dy*y0 + dz*z0)**2 - c1*(x0*x0 + y0*y0 + z0*z0)
```

Comment:
Likewise, this block warrants a comment about how one comes up with the coefficients